aws/resource_aws_dynamodb_global_table.go

```diff
@@ -0,0 +1,330 @@
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsDynamoDbGlobalTable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDynamoDbGlobalTableCreate,
+		Read:   resourceAwsDynamoDbGlobalTableRead,
+		Update: resourceAwsDynamoDbGlobalTableUpdate,
+		Delete: resourceAwsDynamoDbGlobalTableDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(1 * time.Minute),
+			Delete: schema.DefaultTimeout(1 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsDynamoDbGlobalTableName,
+			},
+
+			"replica": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceAwsDynamoDbGlobalTableReplicaHash,
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsDynamoDbGlobalTableCreate(d *schema.ResourceData, meta interface{}) error {
+	dynamodbconn := meta.(*AWSClient).dynamodbconn
+
+	globalTableName := d.Get("name").(string)
+
+	input := &dynamodb.CreateGlobalTableInput{
+		GlobalTableName:  aws.String(globalTableName),
+		ReplicationGroup: expandAwsDynamoDbReplicas(d.Get("replica").(*schema.Set).List()),
+	}
+
+	log.Printf("[DEBUG] Creating DynamoDB Global Table: %#v", input)
+	_, err := dynamodbconn.CreateGlobalTable(input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(globalTableName)
+
+	log.Println("[INFO] Waiting for DynamoDB Global Table to be created")
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			dynamodb.GlobalTableStatusCreating,
+			dynamodb.GlobalTableStatusDeleting,
+			dynamodb.GlobalTableStatusUpdating,
+		},
+		Target: []string{
+			dynamodb.GlobalTableStatusActive,
+		},
+		Refresh:    resourceAwsDynamoDbGlobalTableStateRefreshFunc(d, meta),
+		Timeout:    d.Timeout(schema.TimeoutUpdate),
+		MinTimeout: 10 * time.Second,
+	}
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return err
+	}
+
+	return resourceAwsDynamoDbGlobalTableRead(d, meta)
+}
+
+func resourceAwsDynamoDbGlobalTableRead(d *schema.ResourceData, meta interface{}) error {
+	globalTableDescription, err := resourceAwsDynamoDbGlobalTableRetrieve(d, meta)
+
+	if err != nil {
+		return err
+	}
+	if globalTableDescription == nil {
+		d.SetId("")
+		return nil
+	}
+
+	return flattenAwsDynamoDbGlobalTable(d, globalTableDescription)
```

Comment:
Is there a reason to have this as a separated function? I think it would be more consistent with other resources to keep setting attributes in the read function. What do you think?

Reply:
This is so we can reuse this code support a `aws_dynamodb_global_table` data source in the future, if we would like.